community/values/src/main/java/org/neo4j/values/ValueTuple.java

```diff
@@ -31,6 +33,18 @@ public static ValueTuple of( Value... values )
         return new ValueTuple( values );
     }
 
+    public static ValueTuple of( Object... objects )
+    {
+        assert objects.length > 0 : "Empty ValueTuple is not allowed";
+        assert noNulls( objects );
+        Value[] values = new Value[objects.length];
+        for ( int i = 0; i < values.length; i++ )
+        {
+            values[i] = Values.of( objects[i] );
+        }
+        return new ValueTuple( values );
+    }
+
     private final Value[] values;
 
     private ValueTuple( Value[] values )
```

Comment:
Not really... I added `getOnlyValue` only to make life easier in the parts of the code base that still do not support composite indexes. By calling that method we are assuming the tuple only contains one value, which is checked by `assert values.length == 1`. Calling the method `getFirstValue` would communicate that it should work even for cases when there's more than 1 value in the tuple, which it shouldn't. In these cases we should surely consider all the values!

Or have I misunderstood you?